### DISCUSSION TITLE
Improve UX regarding adding files in new (nonexistent) folders

### DISCUSSION THREAD

**Discussion by inputusernamehere** [ionide/ionide-vscode-fsharp#1639] [Ideas]
Thank you for your work on Ionide! I have a minor improvement suggestion:

So right now it's possible to create a new folder in the regular file explorer, then switch to the F#/Ionide view, and create a new file in that folder. Let's say I call the folder `new_folder`, and input `new_folder/new_file.fs` when I create a new file. It will work as expected.

However if I do not create the folder my self first, I get hit with this error:

Error running command fsharp.explorer.addFile: Could not find a part of the path 'd:\code\ionide-folder-test\new_folder\new_file.fs'.. This is likely caused by the extension that contributes fsharp.explorer.addFile.

I think it would be smoother if Ionide just created the new folder for me.

**Comment by baronfel** [comment]
I made an issue out of this discussion [here](https://github.com/ionide/ionide-vscode-fsharp/issues/1640) because I think the fix is quite straightforward.
